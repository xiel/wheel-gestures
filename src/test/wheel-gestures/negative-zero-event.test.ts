import winRightFast from '../fixtures/negative-zero-event/win-chrome-fast-right-ptp.json'
import winRight from '../fixtures/negative-zero-event/win-chrome-right-ptp.json'
import { recordPhases } from '../helper/recordPhases'

describe('negative zero event', () => {
  test('Win - Chrome - swipe right', () => {
    expect(winRight.wheelEvents.filter((eventData) => Object.is(-0, eventData.deltaX)).length).toBe(1)
    expect(recordPhases(winRight.wheelEvents)).toMatchSnapshot()
  })

  test('Win - Chrome - swipe right fast', () => {
    expect(winRightFast.wheelEvents.filter((eventData) => Object.is(-0, eventData.deltaX)).length).toBe(1)
    expect(recordPhases(winRightFast.wheelEvents)).toMatchSnapshot()
  })
})
