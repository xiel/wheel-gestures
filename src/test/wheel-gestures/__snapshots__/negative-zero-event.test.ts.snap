// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`negative zero event Win - Chrome - swipe right 1`] = `
Object {
  "phases": Array [
    Object {
      "axisDelta": Array [
        0,
        0,
        0,
      ],
      "axisMovement": Array [
        -1810.5,
        0,
        0,
      ],
      "axisVelocity": Array [
        -0.027716186256142486,
        0,
        0,
      ],
      "event": Object {
        "deltaMode": 0,
        "deltaX": -1.5,
        "deltaY": 0,
        "timeStamp": 5711.669999989681,
      },
      "hasPrevious": true,
      "isEnding": true,
      "isMomentum": true,
      "isMomentumCancel": false,
      "isStart": false,
      "range": Array [
        95,
        95,
      ],
      "wheelType": "momentum",
    },
  ],
}
`;

exports[`negative zero event Win - Chrome - swipe right fast 1`] = `
Object {
  "phases": Array [
    Object {
      "axisDelta": Array [
        0,
        0,
        0,
      ],
      "axisMovement": Array [
        -3631.5,
        0,
        0,
      ],
      "axisVelocity": Array [
        -0.032752879522086574,
        0,
        0,
      ],
      "event": Object {
        "deltaMode": 0,
        "deltaX": -1.5,
        "deltaY": 0,
        "timeStamp": 40598.1900000188,
      },
      "hasPrevious": true,
      "isEnding": true,
      "isMomentum": true,
      "isMomentumCancel": false,
      "isStart": false,
      "range": Array [
        107,
        107,
      ],
      "wheelType": "momentum",
    },
  ],
}
`;
